config/routes.rb

```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'upload/' => 'upload#index', as: 'upload'
```

Comment:
Use the `new` method in `track` controller. Also you can remove the / after `upload`.